uberfire-project/uberfire-project-backend/src/test/java/org/guvnor/common/services/project/backend/server/AbstractModuleServiceTest.java

```diff
@@ -134,4 +142,43 @@ public void testRenameEventFiredBeforeDeleteEvent() {
         inOrder.verify(renameProjectEvent).fire(any());
         inOrder.verify(ioService).endBatch();
     }
+
+    @Test
+    public void testUseRepoServiceToDeleteRootModule() {
+        when(path.getFileName()).thenReturn("pom.xml");
+        when(path.toURI()).thenReturn("file:///space/project1/pom.xml");
+        when(resourceResolver.resolveModule(any(Path.class))).thenReturn(module);
+        when(pomService.load(any())).thenReturn(mock(POM.class));
+        Repository repo = mock(Repository.class);
+        when(repoService.getRepository(eq(org.uberfire.backend.server.util.Paths.convert(Paths.get("file:///space/project1"))))).thenReturn(repo);
+        String alias = "repo-alias";
+        when(repo.getAlias()).thenReturn(alias);
+        Space space = new Space("space");
+        when(repo.getSpace()).thenReturn(space);
+
+        abstractProjectService.delete(path, "");
+
+        verify(repoService).removeRepository(eq(space), eq(alias));
+        verify(ioService, times(0)).delete(any(), any());
+    }
+
+    @Test
+    public void testUseIOServiceToDeleteRootModule() {
```

Comment:
I think the name should be `testUseIOServiceToDeleteSubModule` (or something like that) instead.

Reply:
Oh yeah, good catch.